### DISCUSSION TITLE
Use Rails' `:as` option to name path helpers.

### DISCUSSION THREAD

**Discussion by kstevens715** [ElMassimo/js_from_routes#37] [Ideas]
The following routes are a slight modification to the [routes file checked into this repo](https://github.com/ElMassimo/js_from_routes/blob/main/playground/vanilla/config/routes.rb).

Currently, these two routes only produce a single path helper because [routes with the same `action` are uniqued](https://github.com/ElMassimo/js_from_routes/blob/d421dc6339acbbccc5b238e8677bfa932da5637d/js_from_routes/lib/js_from_routes/generator.rb#L18). `js_from_routes` could produce helpers for both routes if it made use of [Rails' `:as` option](https://guides.rubyonrails.org/routing.html#overriding-the-named-route-helpers). This makes a lot of sense to me because that is the way Rails provides to make custom names for the path helpers inside of Rails. This would allow those same path helpers to be exposed to JS.

Using the `:as` option to name the helpers would probably be good even outside of this example that currently causes a path helper to not get produced.

```ruby
resources :user_preferences, only: [], export: true do
  get "/switch_to_classic", action: :switch_to_classic, on: :collection
  get "/switch_to_classic/option_with_args/:id",
    action: :switch_to_classic,
    on: :collection,
    as: :switch_to_classic_with_args
end
```

Given the above Rails routes, I would expect the exported JS path helpers to be something like:

```js
export default {
  switch_to_classic: definePathHelper('get', '/switch_to_classic'),
  switch_to_classic_with_args: definePathHelper('get', '/switch_to_classic/option_with_args/:id'),
}
```

**Comment by ElMassimo** [comment]
Hi Kyle!

I like this proposal, if you come up with a way to obtain the original `as`, PRs are welcome.

Unfortunately, `as` is [discarded](https://github.com/rails/rails/blob/5726b1d1d7665c33830ad114f45258ecf772b7b6/actionpack/lib/action_dispatch/routing/mapper.rb#L1975-L1979), and I don't know of a reliable way to reconstruct it from the information in an individual route object.

Perhaps this could be implemented using a new `export_as` option, but I'd like to see how common this scenario is before adding a new option.
